extensions/amp-geo/0.1/amp-geo.js

```diff
@@ -208,34 +208,54 @@ export class AmpGeo extends AMP.BaseElement {
       self.findCountry_(doc);
       self.matchCountryGroups_(config);
 
+      let classesToRemove = [];
+
       switch (self.mode_) {
         case mode.GEO_OVERRIDE:
-          self.clearPreRender_(doc);
+          classesToRemove = self.clearPreRender_(doc);
           // Intentionally fall through.
         case mode.GEO_HOT_PATCH:
           // Build the AMP State, add classes
           states.ISOCountry = self.country_;
 
-          for (let group = 0; group < self.matchedGroups_.length; group++) {
-            doc.body.classList.add(GROUP_PREFIX + self.matchedGroups_[group]);
-            states[self.matchedGroups_[group]] = true;
-          }
-          doc.body.classList.add(COUNTRY_PREFIX + this.country_);
-          doc.body.classList.remove('amp-geo-pending');
+          const classesToAdd = self.matchedGroups_.map(group => {
+            states[group] = true;
+            return GROUP_PREFIX + group;
+          });
+
           states.ISOCountryGroups = self.matchedGroups_;
+          classesToAdd.push(COUNTRY_PREFIX + this.country_);
+
+          // Let the runtime know we're mutating the doc.body
+          // Actual change happens in callback to runtime can
+          // optimize dom mutations.
+          self.mutateElement(() => {
+            const classList = doc.body.classList;
+            // Always remove the pending class
+            classesToRemove.push('amp-geo-pending');
+            classList.remove.apply(classList, classesToRemove);
```

Comment:
Eek, just remembered that IE doesn't support multi-parameter `add` and `remove`: https://developer.mozilla.org/en-US/docs/Web/API/Element/classList#Browser_compatibility